app/lib/actions/katello/repository/add_export_distributor.rb

```diff
@@ -0,0 +1,16 @@
+module Actions
+  module Katello
+    module Repository
+      class AddExportDistributor < Actions::Base
+        def plan(repo)
+          ::User.current = ::User.anonymous_admin
+          repo = ::Katello::Repository.find(repo.id)
+          plan_action(Pulp::Repository::AssociateDistributor,
+                      repo_id: repo[:pulp_id],
+                      config: {:http => true, :https => true},
+                      type_id: 'export_distributor')
+        end
```

Comment:
Would I need to create the export_distributor directly in Pulp and then run the refresh? I'm not sure how the distributor list would get updated for `export_distributor` otherwise.

Right now I have this in my migration task, which only fires AddExportDistributor if it's not already there:

```ruby
            if not repo.distributors.any? { |r| r[:distributor_type_id] == 'export_distributor'}
              ForemanTasks.sync_task(::Actions::Katello::Repository::AddExportDistributor, repo)
            end
```